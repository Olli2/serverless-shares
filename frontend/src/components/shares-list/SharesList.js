import React from 'react'
import {Share} from '../share/Share'

export const SharesList = ({ shares }) => {
    return (
        <div className="shares-list shares-list--animated">
            {shares.map(share => (
                <Share share={share} key={share.id}/>                
            ))}
        </div>
    )
}