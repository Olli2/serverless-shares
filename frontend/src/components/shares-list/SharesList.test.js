import React from 'react';
import {render} from '@testing-library/react';
import {SharesList} from './SharesList';

jest.mock('../share/Share', () => ({
    Share: jest.fn((obj) => {
        return `ShareMock-${obj.share.id}`
    })
}))

const mockShares = [
    {
        id: 1
    },
    {
        id: 2
    },
    {
        id: 3
    },
]

describe('Shares list component', () => {
    it('should match with snapshot', () => {
        const actual = render(<SharesList shares={mockShares}/>)
        expect(actual.asFragment).toMatchSnapshot()
    })

    it('should map all objects', () => {
        const actual = render(<SharesList shares={mockShares}/>)
        expect(actual.container).toHaveTextContent('ShareMock-1')
        expect(actual.container).toHaveTextContent('ShareMock-2')
        expect(actual.container).toHaveTextContent('ShareMock-3')
    })
})