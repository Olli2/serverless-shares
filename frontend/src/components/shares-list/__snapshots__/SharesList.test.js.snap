// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shares list component should match with snapshot 1`] = `[Function]`;
